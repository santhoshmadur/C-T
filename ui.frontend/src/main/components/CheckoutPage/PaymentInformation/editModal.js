/*******************************************************************************
 *
 *    Copyright 2020 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React, { useMemo } from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { bool, func } from 'prop-types';

import { useEditModal } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/useEditModal';
import Dialog from '@magento/venia-ui/lib/components/Dialog';
import editModalOperations from '@magento/venia-ui/lib/components/CheckoutPage/PaymentInformation/editModal.gql';
import editablePayments from './editablePaymentCollection';

const EditModal = props => {
    const { onClose, isOpen } = props;
    const { formatMessage } = useIntl();

    const talonProps = useEditModal({ onClose, ...editModalOperations });

    const {
        selectedPaymentMethod,
        handleUpdate,
        handleClose,
        handlePaymentSuccess,
        handlePaymentReady,
        updateButtonClicked,
        resetUpdateButtonClicked,
        handlePaymentError
    } = talonProps;

    const paymentMethodComponent = useMemo(() => {
        const isEditable = Object.keys(editablePayments).includes(selectedPaymentMethod);
        if (isEditable) {
            const PaymentMethodComponent = editablePayments[selectedPaymentMethod];
            return (
                <PaymentMethodComponent
                    onPaymentReady={handlePaymentReady}
                    onPaymentSuccess={handlePaymentSuccess}
                    onPaymentError={handlePaymentError}
                    resetShouldSubmit={resetUpdateButtonClicked}
                    shouldSubmit={updateButtonClicked}
                />
            );
        } else {
            return (
                <div>
                    <FormattedMessage
                        id={'checkoutPage.paymentMethodStatus'}
                        defaultMessage={'The selected method is not supported for editing.'}
                        values={{ selectedPaymentMethod }}
                    />
                </div>
            );
        }
    }, [
        handlePaymentError,
        handlePaymentReady,
        handlePaymentSuccess,
        resetUpdateButtonClicked,
        selectedPaymentMethod,
        updateButtonClicked
    ]);

    return (
        <Dialog
            confirmText={'Update'}
            confirmTranslationId={'global.updateButton'}
            isOpen={isOpen}
            onCancel={handleClose}
            onConfirm={handleUpdate}
            shouldDisableAllButtons={updateButtonClicked}
            shouldDisableConfirmButton={updateButtonClicked}
            title={formatMessage({
                id: 'checkoutPage.editPaymentInformation',
                defaultMessage: 'Edit Payment Information'
            })}
        >
            {paymentMethodComponent}
        </Dialog>
    );
};

export default EditModal;

EditModal.propTypes = {
    onClose: func.isRequired,
    isOpen: bool
};
