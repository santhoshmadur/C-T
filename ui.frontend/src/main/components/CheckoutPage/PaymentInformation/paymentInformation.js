/*******************************************************************************
 *
 *    Copyright 2020 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React, { Suspense } from 'react';
import { FormattedMessage } from 'react-intl';
import { Form } from 'informed';
import { shape, func, string, bool, instanceOf } from 'prop-types';

import { usePaymentInformation } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentInformation';
import CheckoutError from '@magento/peregrine/lib/talons/CheckoutPage/CheckoutError';

import { useStyle } from '@magento/venia-ui/lib/classify';
import paymentInformationOperations from '@magento/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentInformation.gql';
import defaultClasses from '@magento/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentInformation.css';
import LoadingIndicator from '@magento/venia-ui/lib/components/LoadingIndicator';

const PaymentMethods = React.lazy(() => import('./paymentMethods'));
const EditModal = React.lazy(() => import('./editModal'));
const Summary = React.lazy(() => import('./summary'));

const PaymentInformation = props => {
    const { classes: propClasses, onSave, resetShouldSubmit, setCheckoutStep, shouldSubmit, checkoutError } = props;

    const classes = useStyle(defaultClasses, propClasses);

    const talonProps = usePaymentInformation({
        onSave,
        checkoutError,
        resetShouldSubmit,
        setCheckoutStep,
        shouldSubmit,
        ...paymentInformationOperations
    });

    const {
        doneEditing,
        handlePaymentError,
        handlePaymentSuccess,
        hideEditModal,
        isLoading,
        isEditModalActive,
        showEditModal
    } = talonProps;

    if (isLoading) {
        return (
            <LoadingIndicator classes={{ root: classes.loading }}>
                <FormattedMessage
                    id={'checkoutPage.loadingPaymentInformation'}
                    defaultMessage={'Fetching Payment Information'}
                />
            </LoadingIndicator>
        );
    }

    const paymentInformation = doneEditing ? (
        <Summary onEdit={showEditModal} />
    ) : (
        <Form>
            <PaymentMethods
                onPaymentError={handlePaymentError}
                onPaymentSuccess={handlePaymentSuccess}
                resetShouldSubmit={resetShouldSubmit}
                shouldSubmit={shouldSubmit}
            />
        </Form>
    );

    const editModal = doneEditing ? (
        <Suspense fallback={null}>
            <EditModal onClose={hideEditModal} isOpen={isEditModalActive} />
        </Suspense>
    ) : null;

    return (
        <div className={classes.root}>
            <div className={classes.payment_info_container}>
                <Suspense fallback={null}>{paymentInformation}</Suspense>
            </div>
            {editModal}
        </div>
    );
};

export default PaymentInformation;

PaymentInformation.propTypes = {
    classes: shape({
        container: string,
        payment_info_container: string,
        review_order_button: string
    }),
    onSave: func.isRequired,
    checkoutError: instanceOf(CheckoutError),
    resetShouldSubmit: func.isRequired,
    shouldSubmit: bool,
    setCheckoutStep: func
};
