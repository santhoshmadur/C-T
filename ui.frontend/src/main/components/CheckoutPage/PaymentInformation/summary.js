/*******************************************************************************
 *
 *    Copyright 2020 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React from 'react';
import { FormattedMessage } from 'react-intl';
import { shape, string, func } from 'prop-types';

import { useSummary } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/useSummary';
import { useStyle } from '@magento/venia-ui/lib/classify';

import defaultClasses from '@magento/venia-ui/lib/components/CheckoutPage/PaymentInformation/summary.css';
import LoadingIndicator from '@magento/venia-ui/lib/components/LoadingIndicator';
import summaryPayments from './summaryPaymentCollection';

const Summary = props => {
    const { classes: propClasses, onEdit } = props;
    const classes = useStyle(defaultClasses, propClasses);

    const talonProps = useSummary();

    const { isLoading, selectedPaymentMethod } = talonProps;

    if (isLoading && !selectedPaymentMethod) {
        return (
            <LoadingIndicator classes={{ root: classes.loading }}>
                <FormattedMessage
                    id={'checkoutPage.loadingPaymentInformation'}
                    defaultMessage={'Fetching Payment Information'}
                />
            </LoadingIndicator>
        );
    }

    const hasCustomSummaryComp = Object.keys(summaryPayments).includes(selectedPaymentMethod.code);

    if (hasCustomSummaryComp) {
        const SummaryPaymentMethodComponent = summaryPayments[selectedPaymentMethod.code];
        return <SummaryPaymentMethodComponent onEdit={onEdit} />;
    } else {
        return (
            <div className={classes.root}>
                <div className={classes.heading_container}>
                    <h5 className={classes.heading}>
                        <FormattedMessage
                            id={'checkoutPage.paymentInformation'}
                            defaultMessage={'Payment Information'}
                        />
                    </h5>
                </div>
                <div className={classes.card_details_container}>
                    <span className={classes.payment_details}>{selectedPaymentMethod.title}</span>
                </div>
            </div>
        );
    }
};

export default Summary;

Summary.propTypes = {
    classes: shape({
        root: string,
        heading_container: string,
        heading: string,
        card_details_container: string,
        payment_details: string
    }),
    onEdit: func.isRequired
};
