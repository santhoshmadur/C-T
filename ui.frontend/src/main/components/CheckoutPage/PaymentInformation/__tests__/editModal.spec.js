/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
/* eslint-disable react/display-name */
import React from 'react';
import render from '../../../utils/test-utils';
import { useEditModal } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/useEditModal';
import EditModal from '../editModal';

jest.mock('@magento/venia-ui/lib/classify');
jest.mock('@magento/venia-ui/lib/components/Dialog', () => props => (
    <mock-Dialog {...props}>{props.children}</mock-Dialog>
));
jest.mock('@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/useEditModal', () => ({
    useEditModal: jest.fn().mockReturnValue({
        selectedPaymentMethod: 'braintree',
        isLoading: false,
        updateButtonClicked: false,
        handleClose: jest.fn(),
        handlePaymentError: jest.fn(),
        handlePaymentReady: jest.fn(),
        handlePaymentSuccess: jest.fn(),
        handleUpdate: jest.fn(),
        resetUpdateButtonClicked: jest.fn()
    })
}));

jest.mock('../editablePaymentCollection', () => ({
    braintree: props => <mock-Braintree id={'BraintreeMockId'} {...props} />
}));

jest.mock('@magento/venia-ui/lib/components/Button', () => {
    return props => <mock-Button {...props} />;
});

jest.mock('@magento/venia-ui/lib/components/Icon', () => {
    return props => <mock-Icon {...props} />;
});

test('Should return correct shape', () => {
    const tree = render(<EditModal />);

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Should render creditCard component if selectedPaymentMethod is braintree', () => {
    useEditModal.mockReturnValueOnce({
        selectedPaymentMethod: 'braintree',
        isLoading: true,
        updateButtonClicked: false,
        handleClose: jest.fn(),
        handleUpdate: jest.fn(),
        handlePaymentSuccess: jest.fn(),
        handlePaymentReady: jest.fn()
    });

    const tree = render(<EditModal />);

    expect(() => {
        tree.root.findByProps({ id: 'BraintreeMockId' });
    }).not.toThrow();
});

test('Should not render creditCard component if selectedPaymentMethod is not braintree', () => {
    useEditModal.mockReturnValueOnce({
        selectedPaymentMethod: 'paypal',
        isLoading: true,
        updateButtonClicked: false,
        handleClose: jest.fn(),
        handleUpdate: jest.fn(),
        handlePaymentSuccess: jest.fn(),
        handlePaymentReady: jest.fn()
    });

    const tree = render(<EditModal />);

    expect(() => {
        tree.root.findByProps({ id: 'BraintreeMockId' });
    }).toThrow('No instances found with props: {"id":"BraintreeMockId"}');
});
