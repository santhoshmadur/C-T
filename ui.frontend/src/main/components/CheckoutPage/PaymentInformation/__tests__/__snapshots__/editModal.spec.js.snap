// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return correct shape 1`] = `
<mock-Dialog
  confirmText="Update"
  confirmTranslationId="global.updateButton"
  onCancel={[MockFunction]}
  onConfirm={[MockFunction]}
  shouldDisableAllButtons={false}
  shouldDisableConfirmButton={false}
  title="Edit Payment Information"
>
  <mock-Braintree
    id="BraintreeMockId"
    onPaymentError={[MockFunction]}
    onPaymentReady={[MockFunction]}
    onPaymentSuccess={[MockFunction]}
    resetShouldSubmit={[MockFunction]}
    shouldSubmit={false}
  />
</mock-Dialog>
`;
